src/ruby.rs

```diff
@@ -741,4 +742,57 @@ tests! {
             "__test_define_module_function_arity_3__ works!".to_ruby()
         );
     }
+
+    #[test]
+    fn test_undef_method(assert: &mut Assertions) {
+        let class = unsafe {
+            rb_define_class(
+                cstr!("TestUndefClass"),
+                rb_cObject
+            )
+        };
+
+        unsafe { rb_undef_method(class, cstr!("to_s")) };
+
+        assert.rb_eq(
+            lazy_eval("::TestUndefClass.new.respond_to?(:to_s)"),
+            unsafe { Qfalse }
+        );
```

Comment:
Should we also check calling the method gives you a NoMethodError? Something like

```
assert.rb_eq(
  lazy_eval(r#"
    exception = begin
      TestUndefClass.new.to_s
    rescue => e
      e
    end

    exception.class
  "),
  rb_eNoMethodError
)
```

There are probably terser ways to write it, but that's the idea